src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
     Job.Runner.addToilOptions(parser_run)
     options = parser.parse_args()
 
     cwd = os.getcwd()
     if options.command == 'generate-config' or options.command == 'generate':
-        generate_file(os.path.join(cwd, 'gatk-preprocessing.config'), generate_config)
+        generate_file(os.path.join(cwd, 'config-toil-germline.yaml'), generate_config)
     if options.command == 'generate-manifest' or options.command == 'generate':
-        generate_file(os.path.join(cwd, 'gatk-preprocessing-manifest.tsv'), generate_manifest)
-
+        generate_file(os.path.join(cwd, 'manifest-toil-germline.tsv'), generate_manifest)
     # Pipeline execution
     elif options.command == 'run':
+        # Program checks
+        for program in ['curl', 'docker']:
+            require(next(which(program)), program + ' must be installed on every node.'.format(program))
+
         require(os.path.exists(options.config), '{} not found. Please run '
-                                             '"generate-config"'.format(options.config))
-        if not options.sample:
-            require(os.path.exists(options.manifest), '{} not found and no sample provided. Please '
-                                                       'run "generate-manifest"'.format(options.manifest))
-        # Parse config
-        parsed_config = {x.replace('-', '_'): y for x, y in yaml.load(open(options.config).read()).iteritems()}
-        inputs = argparse.Namespace(**parsed_config)
+                                                '"generate-config"'.format(options.config))
+
+        require(options.manifest or options.sample, 'Must provide path to manifest or '
+                                                    'sample information at the command line')
+
+        # Obtain list of samples
+        samples = []
         if options.manifest:
-            inputs.manifest = options.manifest
-    
-    inputs.cpu_count = multiprocessing.cpu_count() # FIXME: should not be called from toil-leader, see #186
-    inputs.memory = '15'
-    
-    Job.Runner.startToil(Job.wrapJobFn(batch_start, inputs, options.sample, options.output_dir, options.suffix), options)
+            samples.extend(parse_manifest(options.manifest))
+
+        if options.sample:
+            uuid, url = options.sample
+            samples.append((uuid, url, None))
+
+        if len(samples) == 0:
+            raise ValueError('No samples were detected in the manifest or on the command line')
+
+        # Parse config
+        config = {x.replace('-', '_'): y for x, y in yaml.load(open(options.config).read()).iteritems()}
+        config_fields = set(config)
+        required_fields = {'genome_fasta', 'output_dir', 'file_size', 'run_bwa', 'preprocess',
+                           'run_vqsr', 'joint', 'xmx'}
+        require(config_fields > required_fields,
+                'Missing config parameters:\n{}'.format(', '.join(required_fields - config_fields)))
+
+        config['file_size'] = human2bytes(config['file_size'])
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.